database.go

```diff
@@ -0,0 +1,26 @@
+package eremetic
+
+import "encoding/json"
+
+// TaskDB defines the functions needed by the database abstraction layer
+type TaskDB interface {
+	Clean() error
+	Close()
+	PutTask(task *Task) error
+	ReadTask(id string) (Task, error)
+	ReadUnmaskedTask(id string) (Task, error)
+	ListNonTerminalTasks() ([]*Task, error)
+}
+
+const Masking = "*******"
+
+func ApplyMask(task *Task) {
```

Comment:
When in the eremetic namespace these names are a bit unclear. Maybe we should include task somewhere in the name? Or make these methods on the task struct?